src/isomorphic/React.js

```diff
@@ -86,4 +86,31 @@ var React = {
 
 };
 
+if (__DEV__) {
+  // React.DOM factories are deprecated. Set up a Proxy to listen for
+  // accesses to the React.DOM namespace and alert users to switch to
+  // the `react-addons-dom-factories` package.
+  if (typeof Proxy === 'function') {
+    var hasWarnedOfFactories = false;
```

Comment:
Maybe we should instead use `Object.defineProperty` to define getters for each method instead, so we don't have to rely on `Proxy` at all and split up the logic for these warnings.